dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/http2/GenericHttp2ServerTransportListener.java

```diff
@@ -187,6 +187,12 @@ protected final Http2ServerChannelObserver getServerChannelObserver() {
         return serverChannelObserver;
     }
 
+    @Override
+    public void onStreamClosed() {
+        // doing on event loop thread
+        getStreamingDecoder().close();
```

Comment:
**Suggestion**: Validate the state of getStreamingDecoder() before invoking close() to avoid potential NullPointerException. <br> **Code Suggestion**: 
 ```
 +if (getStreamingDecoder() != null) {
 +    getStreamingDecoder().close();
 +}
 ```